Content.Shared/_Lavaland/Chasm/PreventChasmFallingSystem.cs

```diff
@@ -0,0 +1,112 @@
+// SPDX-FileCopyrightText: 2025 deltanedas <39013340+deltanedas@users.noreply.github.com>
+// SPDX-FileCopyrightText: 2025 deltanedas <@deltanedas:kde.org>
+// SPDX-FileCopyrightText: 2025 gluesniffler <159397573+gluesniffler@users.noreply.github.com>
+// SPDX-FileCopyrightText: 2025 gluesniffler <linebarrelerenthusiast@gmail.com>
+// SPDX-FileCopyrightText: 2025 pheenty <fedorlukin2006@gmail.com>
+// SPDX-FileCopyrightText: 2025 username <113782077+whateverusername0@users.noreply.github.com>
+// SPDX-FileCopyrightText: 2025 whateverusername0 <whateveremail>
+//
+// SPDX-License-Identifier: AGPL-3.0-or-later
+
+using Content.Shared.Chasm;
+using Content.Shared.Interaction;
+using Content.Shared.Inventory;
+using Content.Shared.Timing;
+using Robust.Shared.Audio.Systems;
+using Robust.Shared.Containers;
+using Robust.Shared.Map;
+using Robust.Shared.Random;
+
+namespace Content.Shared._Lavaland.Chasm;
+
+public sealed class PreventChasmFallingSystem : EntitySystem
+{
+    [Dependency] private readonly SharedAudioSystem _audio = default!;
+    [Dependency] private readonly SharedTransformSystem _transform = default!;
+    [Dependency] private readonly IRobustRandom _random = default!;
+    [Dependency] private readonly SharedInteractionSystem _interaction = default!;
+    [Dependency] private readonly EntityLookupSystem _lookup = default!;
+    [Dependency] private readonly UseDelaySystem _delay = default!;
+
+    public override void Initialize()
+    {
+        base.Initialize();
+
+        SubscribeLocalEvent<PreventChasmFallingComponent, BeforeChasmFallingEvent>(OnBeforeFall);
+        SubscribeLocalEvent<InventoryComponent, BeforeChasmFallingEvent>(Relay);
+    }
+
+    private void OnBeforeFall(EntityUid uid, PreventChasmFallingComponent comp, ref BeforeChasmFallingEvent args)
+    {
+        if (TryComp<UseDelayComponent>(uid, out var useDelay) && _delay.IsDelayed((uid, useDelay)))
+            return;
+
+        args.Cancelled = true;
+        var coordsValid = false;
+        var coords = Transform(args.Entity).Coordinates;
+
+        const int attempts = 20;
+        var curAttempts = 0;
+        while (!coordsValid)
+        {
+            curAttempts++;
+            if (curAttempts > attempts)
+                return; // Just to be safe from stack overflow
+
+            var newCoords = new EntityCoordinates(Transform(args.Entity).ParentUid, coords.X + _random.NextFloat(-5f, 5f), coords.Y + _random.NextFloat(-5f, 5f));
+            if (!_interaction.InRangeUnobstructed(args.Entity, newCoords, -1f) ||
+                _lookup.GetEntitiesInRange<ChasmComponent>(newCoords, 1f).Count > 0)
+                continue;
+
+            _transform.SetCoordinates(args.Entity, newCoords);
+            _transform.AttachToGridOrMap(args.Entity, Transform(args.Entity));
+            _audio.PlayPvs("/Audio/Items/Mining/fultext_launch.ogg", args.Entity);
```

Comment:
`'ResolvedSoundSpecifier.implicit operator ResolvedSoundSpecifier(string)' is obsolete: 'String literals for sounds are deprecated, use a SoundSpecifier or ResolvedSoundSpecifier as appropriate instead'`
```suggestion
            _audio.PlayPvs(new ResolvedPathSpecifier("/Audio/Items/Mining/fultext_launch.ogg"), args.Entity);
```